### ISSUE TITLE
If all enum values are handled in a switch it should not create a waning

### ISSUE THREAD

**Issue by zoechi** [google/tuneup.dart#26]
This function declares a return type of 'CommitRequestMode', but does not end with a return statement at lib/src/convert_to_datastore.dart, line 23.

```Dart
  enum CommitRequestMode { transactional, nonTransactional }

  String commitRequestMode(
      model.CommitRequestMode mode) {
    switch (mode) {
      case model.CommitRequestMode.nonTransactional:
        return 'NON_TRANSACTIONAL';
      case model.CommitRequestMode.transactional:
        return 'TRANSACTIONAL';
    }
  }
```

How is this package related to https://github.com/dart-lang/linter? Seems to have a large overlapping set of features.

**Comment by devoncarew** [comment]
Thanks Günter, can you re-file this at dartbug.com/new? The fix for this will need to go into the analysis engine.

As I understand it, linter `will` provide several optional analysis on top of what's provided by the analysis engine. Like code health rather than code correctness.

This package is more a set of utilities I'd written and found useful for my own development. Specifically for `tuneup check`, that runs the analyzer over all the source in a project, rather than just a few libraries. It's really useful in a CI context, to know that no errors introduced by an PRs.

**Comment by zoechi** [comment]
http://dartbug.com/22110 has status fixed therefore I guessed it's a bug in your implementation. Probably just needs an analyzer update then. analyzer 0.24.0 was published yesterday :-)

**Comment by devoncarew** [comment]
published v0.0.5+2!
